router/http/src/test/scala/io/buoyant/router/http/DiscardProxySideEffectsTest.scala

```diff
@@ -0,0 +1,70 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle._
+import com.twitter.finagle.http.{Status, _}
+import com.twitter.util.Future
+import io.buoyant.test.Awaits
+import org.scalatest.FunSuite
+
+class DiscardProxySideEffectsTest extends FunSuite with Awaits {
+
+  def service: Service[Request, Response] = {
+    val svc = Service.mk[Request, Response] { req =>
+      val rsp = Response()
+      rsp.status = Status.Ok
+      rsp.setContentString(req.uri)
+      rsp.headerMap ++= req.headerMap
+      Future.value(rsp)
+    }
+
+    val stk = DiscardProxySideEffects.module.toStack(
+      Stack.Leaf(DiscardProxySideEffects.module.role, ServiceFactory.const(svc))
+    )
+
+    await(
+      stk.make(Stack.Params.empty)()
+    )
+  }
+
+  test("filter rewrites uri, restores host and drops Proxy-* headers for proxied requests") {
+    val req = Request(Method.Get, "http://acme.co:8080/foo;matrix=uri;/bar?x=1&y=2")
+    req.host = "proxy:123"
+    req.headerMap.set("Proxy-Connection", "Keep-Alive")
+    req.headerMap.set("Proxy-Authorization", "Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==")
+    req.headerMap.set("Proxy-Authenticate", "Basic")
+    req.headerMap.set("Proxy-Foo", "Bar")
+    req.headerMap.set("Pragma", "42")
+
+    val result = await(service(req))
+    assert(result.getContentString() == "/foo;matrix=uri;/bar?x=1&y=2")
+    assert(
+      result.headerMap.toSet == Set(
+        ("Host", "acme.co:8080"),
+        ("Pragma", "42")
+      )
+    )
+  }
+
+  test("filter doesn't affect non-proxied requests") {
+    val req = Request(Method.Get, "/foo;matrix=uri;/bar?x=1&y=2")
+    req.host = "acme.co:8080"
+    req.headerMap.set("Proxy-Connection", "Keep-Alive")
+    req.headerMap.set("Proxy-Authorization", "Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==")
+    req.headerMap.set("Proxy-Authenticate", "Basic")
+    req.headerMap.set("Proxy-Foo", "Bar")
+    req.headerMap.set("Pragma", "42")
+
+    val result = await(service(req))
+    assert(result.getContentString() == "/foo;matrix=uri;/bar?x=1&y=2")
+    assert(
+      result.headerMap.toSet == Set(
+        ("Proxy-Connection", "Keep-Alive"),
```

Comment:
Also though about that - will do. :)

So I see to options: 
1) always drop all `Proxy-*` headers
2) always drop `Proxy-Connection` header and deal with other `Proxy-*` headers if we're in 'proxy mode'

what would you prefer?